infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/pvc/PVCJobExecutor.java

```diff
@@ -0,0 +1,240 @@
+   * @see PVCJob
+   */
+  void execute(String workspaceId, PVCJob job, String... jobArguments) {
+    final String podName = job.name + '_' + workspaceId;
+    final String[] command = buildCommand(job.baseCommand, jobArguments);
+    final Pod pod = newPod(podName, command);
+    OpenShiftPods pods = null;
+    try {
+      pods = factory.create(workspaceId).pods();
+      pods.create(pod);
+      final Pod finished = pods.wait(podName, WAIT_POD_TIMEOUT_MIN, POD_PREDICATE::apply);
+      if (POD_PHASE_FAILED.equals(finished.getStatus().getPhase())) {
+        throw new InfrastructureException(
+            format("Job command '%s' execution is failed.", Arrays.toString(command)));
+      }
+    } catch (InfrastructureException ex) {
+      LOG.error(
+          "Unable to perform '" + job.name + "' job for the workspace '" + workspaceId + "'.", ex);
+    } finally {
+      if (pods != null) {
+        try {
+          pods.delete(podName);
+        } catch (InfrastructureException ignored) {
+        }
+      }
+    }
+  }
+
+  /**
+   * Executes the job with the specified arguments asynchronously.
+   *
+   * @see #execute(String, PVCJob, String...)
+   */
+  CompletableFuture<Void> executeAsync(String workspaceId, PVCJob job, String... jobArguments) {
+    return CompletableFuture.runAsync(
+        ThreadLocalPropagateContext.wrap(() -> execute(workspaceId, job, jobArguments)), executor);
+  }
+
+  /**
+   * Builds the command by given base and paths.
+   *
+   * <p>Command is consists of base(e.g. rm -rf) and list of directories which are modified with
+   * mount path.
+   *
+   * @param base command base
+   * @param dirs the paths which are used as arguments for the command base
+   * @return complete command with given arguments
+   */
+  String[] buildCommand(String[] base, String... dirs) {
+    final String[] dirsWithPath =
+        Arrays.stream(dirs)
+            .map(dir -> JOB_MOUNT_PATH + (dir.startsWith("/") ? dir : '/' + dir))
+            .toArray(String[]::new);
+
+    final String[] command = new String[base.length + dirsWithPath.length];
+    System.arraycopy(base, 0, command, 0, base.length);
+    System.arraycopy(dirsWithPath, 0, command, base.length, dirsWithPath.length);
+    return command;
+  }
+
+  /** Checks whether pod is Failed or Successfully finished command execution */
+  static class JobFinishedPredicate implements Predicate<Pod> {
+    @Override
+    public boolean apply(Pod pod) {
+      if (pod.getStatus() == null) {
+        return false;
+      }
+      switch (pod.getStatus().getPhase()) {
+        case POD_PHASE_FAILED:
+          // fall through
+        case POD_PHASE_SUCCEEDED:
+          // job is finished.
+          return true;
+        default:
+          // job is not finished.
+          return false;
+      }
+    }
+  }
+
+  /** Defines PVC job which contains name and base command to execute. */
+  enum PVCJob {
+    CREATE_DIRS("create_dirs", new String[] {"mkdir", "-p"}),
```

Comment:
Consider the usage of specific methods for creation/removal of dirs instead of usage of this enum. I see it more convenient.

Reply:
removed PVCJob, and moved commands into constants